task/net/src/server.rs

```diff
@@ -240,3 +233,343 @@ impl<T: NetServer> idl::InOrderNetImpl for T {
         Ok(out)
     }
 }
+
+pub trait DeviceExt: for<'d> smoltcp::phy::Device<'d> {
+    fn read_and_clear_activity_flag(&self) -> bool;
+
+    fn make_meta(
+        &self,
+        port: u16,
+        size: usize,
+        addr: task_net_api::Address,
+    ) -> UdpMetadata;
+}
+
+/// State for the running network server
+pub struct GenServerImpl<'a, B, E, const N: usize>
+where
+    E: DeviceExt,
+{
+    eth: &'a eth::Ethernet,
+
+    vlan_state: [VLanState<E>; N],
+    client_waiting_to_send: [bool; SOCKET_COUNT],
+    bsp: B,
+
+    mac: EthernetAddress,
+}
+
+struct VLanState<E>
+where
+    E: DeviceExt,
+{
+    socket_handles: [SocketHandle; SOCKET_COUNT],
+    iface: Interface<'static, E>,
+}
+
+impl<'a, B, E, const N: usize> GenServerImpl<'a, B, E, N>
+where
+    B: bsp_support::Bsp,
+    E: DeviceExt,
+{
+    /// Size of buffer that must be allocated to use `dispatch`.
+    pub const INCOMING_SIZE: usize = idl::INCOMING_SIZE;
```

Comment:
Since this isn't server-dependent, we could remove it here (same suggestion as above)

Reply:
Done